fix: CI failures on main branch (Lint Python and Check Documentation Links)

## Problem

The main branch has pre-existing CI failures that are blocking PRs from merging cleanly. Two CI jobs are failing:

1. **Lint Python** - mypy type checking error
2. **Check Documentation Links** - Sphinx documentation build warnings and broken links

## Failing Checks Details

### 1. Lint Python (mypy error)

```
tests/test_drf_integration.py:299: error: Untyped decorator makes function "user_list" untyped  [untyped-decorator]
Found 1 error in 1 file (checked 68 source files)
```

**Root cause**: A decorator in the DRF integration test file is not properly typed, causing mypy to flag the decorated function as untyped.

### 2. Check Documentation Links

Multiple categories of issues:

**a) Duplicate object descriptions in Sphinx autodoc:**
- `ScopedContainer.scope_id`, `ScopedContainer.parent` duplicated
- `dioxide.adapter`, `dioxide.container`, `dioxide.lifecycle` duplicated
- `Profile` enum members duplicated across modules

**b) Documents not included in toctree:**
- `docs/DEVELOPER_EXPERIENCE.md`
- `docs/DOCUMENT_AUDIT.md`
- `docs/DX_VISION.md`
- `docs/GITHUB_PROJECT_SETUP.md`
- `docs/MLP_VISION.md`
- `docs/SPRINT_READY.md`
- `docs/cicd-modernization-plan.md`
- Various ADR files in `docs/design/`

**c) Broken internal cross-references:**
- `../MLP_VISION.md`, `../ROADMAP.md`, `../README` - relative path issues
- `profiles`, `lifecycle`, `framework_integration` - unknown documents

**d) Python filenames parsed as URLs (false positives):**
- `adapters.py`, `services.py`, `apps.py`, `settings.py` in Django integration docs being treated as broken HTTP links

## Acceptance Criteria

- [ ] mypy type checking passes for all files in `python/` and `tests/`
- [ ] Sphinx documentation builds without errors (warnings about toctree are acceptable if intentional)
- [ ] Broken cross-references in documentation are fixed
- [ ] Python filenames in code blocks are not being validated as URLs
- [ ] CI pipeline passes on main branch
- [ ] All PRs can merge cleanly without inheriting pre-existing failures

## Impact

- Blocks all PRs from having clean CI status
- Creates confusion about whether new changes introduce failures
- Undermines confidence in the CI pipeline

## Suggested Fixes

### For mypy error:
- Add type annotation to the decorator in `tests/test_drf_integration.py:299`
- Or add `# type: ignore[untyped-decorator]` if the decorator is from an external library

### For documentation:
- Add `:no-index:` directives to duplicate autodoc entries
- Either add orphan documents to a toctree or mark them as `:orphan:`
- Fix relative path references to use correct Sphinx cross-reference syntax
- Configure linkcheck to skip Python filenames (likely in `conf.py`)

## Related

- CI run: https://github.com/mikelane/dioxide/actions/runs/20412934237

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -125,6 +125,7 @@ myst_enable_extensions = [
 ]
 myst_heading_anchors = 3
 myst_update_mathjax = False
+myst_linkify_fuzzy_links = False  # Only match URLs with schema (http://, https://)
 
 # -- Options for HTML output -------------------------------------------------
 html_theme = 'furo'
@@ -303,15 +304,6 @@ linkcheck_ignore = [
     # External sites that can be slow or unreliable
     r'https://pyo3\.rs.*',
     r'https://www\.maturin\.rs.*',
-    # Local file references misinterpreted as URLs (linkify extension artifact)
-    r'http://README\.md',
-    r'http://COVERAGE\.md',
-    r'http://ROADMAP\.md',
-    r'http://STATUS\.md',
-    r'http://CLAUDE\.md',
-    r'http://CONTRIBUTING\.md',
-    r'http://CHANGELOG\.md',
-    r'http://Criterion\.rs',
 ]
 
 # Anchors to ignore (fragment identifiers that may not exist)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -281,6 +281,7 @@ dev = [
     "django>=4.0.0",
     "djangorestframework>=3.14.0",
     "django-ninja>=1.0.0",
+    "djangorestframework-stubs>=3.16.6",
 ]
 test = [
     "libcst>=1.8.6",

